src/main/java/org/jabref/gui/duplicationFinder/DuplicateSearch.java

```diff
@@ -162,10 +162,8 @@ private void handleDuplicates(DuplicateSearchResult result) {
         final NamedCompound compoundEdit = new NamedCompound(Localization.lang("duplicate removal"));
         // Now, do the actual removal:
         if (!result.getToRemove().isEmpty()) {
-            for (BibEntry entry : result.getToRemove()) {
-                panel.getDatabase().removeEntry(entry);
-                compoundEdit.addEdit(new UndoableRemoveEntry(panel.getDatabase(), entry));
-            }
+            compoundEdit.addEdit(new UndoableRemoveEntries(panel.getDatabase(), new ArrayList<>(result.getToRemove())));
+            panel.getDatabase().removeEntries(new ArrayList<>(result.getToRemove()));
```

Comment:
`Collections.singletonList` should work here, doesn't it?

Reply:
I believe `DuplicateSearch.DuplicateSearchResult.getToRemove` returns a `Collection`, and it could have multiple elements. As of now, I am simply coercing it into a `List`. However, it might make more sense to do the coercion in the method `DuplicateSearch.DuplicateSearchResult.getToRemove` so that it matches `getToAdd`, which returns a `List`. Let me know your thoughts.

Reply:
I just made the change, as I think it's better if `getToRemove` returns a `List`.